producer/dead_letter_producer.go

```
@@ -0,0 +1,70 @@
+/*
+Copyright © 2020 Red Hat, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Package producer contains functions that can be used to produce (that is
+// send) messages to properly configured Kafka broker.
+package producer
+
+import (
+	"github.com/RedHatInsights/insights-results-aggregator/broker"
+	"github.com/Shopify/sarama"
+	"github.com/rs/zerolog/log"
+)
+
+// CreateDeadLetterProducer is a variable for monkey patching in the unit tests
+var CreateDeadLetterProducer = NewDeadLetterProducer
+
+// DeadLetterProducer is a producer for dead letter queue
+type DeadLetterProducer struct {
+	KafkaProducer KafkaProducer
+	Configuration broker.Configuration
+}
+
+// NewDeadLetterProducer constructs producer for payload tracker topic
+func NewDeadLetterProducer(brokerCfg broker.Configuration) (*DeadLetterProducer, error) {
```

do we need to return an error if we return always nil?